main.js

```diff
@@ -31,22 +54,50 @@ function CreateControl(room_name, lights_on, brightness) {
     range.disabled = !lights_on;
     container_b.appendChild(range);
 
-    console.log(room_name + ' lights ' + (lights_on ? 'on' : 'off') + ' @ ' + brightness + '%');
+    console.log(control_name + ' lights ' + (lights_on ? 'on' : 'off') + ' @ ' + brightness + '%');
     return control;
 }
 
+function CreateRoom(room_name) {
+    let room = document.createElement('div');
+    room.classList.add('room');
 
-// -------------------------------------------------------------------------------------------------
+    let text = document.createElement('span');
+    text.classList.add('text');
+    text.innerHTML = room_name;
+    room.appendChild(text);
+
+    return room;
+}
+
+
+// Page builder ------------------------------------------------------------------------------------
 
 
 function ProcessHue(response) {
     let items = JSON.parse(response);
+
+    // controls is a map from control name to HTML element for that control
+    let controls = {};
     for (let i in items) {
         let item = items[i];
         let lights_on = item.state.on;
         let brightness = Math.round(item.state.bri / 254 * 100);
         let control = CreateControl(item.name, lights_on, brightness);
-        document.body.appendChild(control);
+        controls[item.name] = control;
+    }
+
+    // rooms is a map from room name to list of control names in that room
```

Comment:
nit: drop the comment since it's odd to explain here a constant defined somewhere else